project/EnvironmentCheck.scala

```diff
@@ -82,9 +82,6 @@ object EnvironmentCheck {
     (state: State) => {
       val newState = oldTransition(state)
       val logger   = newState.log
-      val graalOk  = graalVersionOk(graalVersion, javaVersion, logger)
```

Comment:
I was just playing with an upgrade of GraalVM - #3663 and got a failure:
```
[error] /engine/runtime/src/main/scala/org/enso/compiler/SerializationManager.scala:54:11: value createSystemThread is not a member of com.oracle.truffle.api.TruffleLanguage.Env
[error]       env.createSystemThread(runnable)
[error]           ^
[error] /engine/runtime/src/main/scala/org/enso/compiler/SerializationManager.scala:44:15: private val env in class SerializationManager is never used
[error]   private val env = compiler.context.getEnvironment
[error]               ^
[error] /engine/runtime/src/main/scala/org/enso/interpreter/runtime/scope/LocalScope.scala:145:40: type Builder is not a member of object com.oracle.truffle.api.frame.FrameDescriptor
[error]     descriptorBuilder: FrameDescriptor.Builder
[error]                                        ^
[error] /engine/runtime/src/main/scala/org/enso/interpreter/runtime/scope/LocalScope.scala:158:45: value newBuilder is not a member of object com.oracle.truffle.api.frame.FrameDescriptor
[error]     val descriptorBuilder = FrameDescriptor.newBuilder()
[error]                                             ^
[error] four errors found
[error] (runtime / Compile / compileIncremental) Compilation failed
```
Obviously caused by me running on 21.3 GraalVM and not GraalVM 22.3.

Looking for the cause I realized the GraalVM check has been disabled by this PR. What was the reason, @hubertp?

Reply:
> Looking for the cause I realized the GraalVM check has been disabled by this PR. What was the reason, @hubertp? CCing @Akirathan

Can't really remember TBH. I probably wouldn't touch it unless I had to, though @JaroslavTulach.
Definitely worth trying to bring it back.